CRM/Core/Smarty/plugins/block.crmPermission.php

```diff
@@ -36,11 +36,11 @@ function smarty_block_crmPermission($params, $content, &$smarty, &$repeat) {
       // This would be due to developer error - better to return nothing to make it more visible.
       return '';
     }
-    $hasPermission = empty($params['has']) || CRM_Core_Permission::check($params['has'], $params['contact_id'] ?? NULL);
+    $hasPermission = empty($params['has']) || CRM_Core_Permission::check(explode(',', $params['has']), $params['contact_id'] ?? NULL);
```

Comment:
I was looking at this closer and have two more comments:

1. If this is intended to implement `OR`, then I don't think it's right because [the syntax](https://github.com/civicrm/civicrm-core/blob/1ef423056ded6da39a257498218c2d682e7247fb/CRM/Core/Permission.php#L81) for check() is wacky. It's currently doing an `AND`.
2. Even if it was doing an OR, I find it unintuitive that `has=A,B` would mean `A or B`, but `not=A,B` would mean "not A and not B", and in fact that's already doable with nested `{crmPermission}` and there is no way to do "not A or not B".

I know I was outvoted on the other PR but I'll vote again for keeping mindbending out of tpl tags, and instead doing anything complex in php and assign a simple variable to the tpl.

Reply:
I would agree that until we really need it, and can agree on a syntax, assigning the permission from php is the path of least resistance.

Reply:
Thanks @demeritcowboy for your feedback. Sorry for missing the AND parameter format in the comment section. Here's a proposal to support OR and AND conjunction for multiple permissions, and not considering multiple conjunctive operators like `a OR (b AND c)`. In these example I am considering '|' as `OR` and ',' as 'AND' conjunctive  operators. Lets say to support - 
1. `A and B and ....` then we could have {crmpermission has='a,b...'} and in PHP the permission parameter should be changes and following format before being used by `CRM_Core_Permission::check` : 
```php
if (strstr($perms, ',')) {
  $perms = explode(',', $perms);
}
```
2. `A or B or ....` then we could have {crmpermission has='a|b|...'} and in PHP 
```php
if (strstr($perms, '|')) {
  $perms = [explode('|', $perms)];
}
```
3. And same goes for `not` parameter Like for `not A or not B` it would be `{crmpermission not='a|b'}`

The intention is to reduce the call for `call_user_func` in tpls and let crmpermission handle multiple permission check, otherwise we would need to have multiple crmpermission tags on tpls for single block or we can leave with `call_user_func(array('CRM_Core_Permission','check')`.

Reply:
@monishdeb I'm suggesting for anything where the tpls get complicated to do like in https://github.com/civicrm/civicrm-core/pull/30436